scipy/linalg/tests/test_lapack.py

```diff
@@ -1666,75 +1666,90 @@ def generate_random_dtype_array(shape, dtype):
+                          *zip(["joba"]*5,
+                               ["'E'", "'F'", "'G'", "'A'", "'R'"]),
+                          *zip(["jobu"]*3, ["'F'", "'W'", "'N'"]),
+                          *zip(["jobv"]*3, ["'J'", "'W'", "'N'"]),
+                          ("jobr", "'N'"),
+                          ("jobt", "'T'"),
+                          ("jobp", "'P'"),
+                          [("jobv", "'N'"), ("jobu", "'N'")]
                           ])
-def test_gejsv_general(size, dtype, gejsv_lambda):
+def test_gejsv_general(size, dtype, job):
     '''
     Tests the lapack routine ?gejsv. This function varies the job? values
-    one at a time per dtype by passing different dtype routines into a lambda
-    that decides parameters. It then tests the general random matrix A of size
-    (m, n) for zero info, that sva, v, and u can be recombined to make A, and
-    that u.T @ u and v.T @ v are the identity. Lastly, we check that iwork [0]
-    and [1] are n.
+    by passing in a single job* and character combination tuple or a list of
+    them. It then tests the general random matrix A of size (m, n) for
+    - sva matches singular values of A
+    - info == 0
+    - that sva, v, and u can be recombined to make A
+    - that u.T @ u and v.T @ v are the identity
+    Lastly, we check that iwork [0] and [1] are the matrix rank.
     '''
-    np.random.seed(42)
-    rtol = 250 * np.finfo(dtype).eps
+    seed(42)
     atol = 100 * np.finfo(dtype).eps
     m, n = size
     A = generate_random_dtype_array((m, n), dtype)
-
     gejsv = get_lapack_funcs('gejsv', dtype=dtype)
-    # pass A and gejsv into lambda expression for parametrized job? values
-    sva, u, v, work, iwork, info = gejsv_lambda(A, gejsv)
+
+    jobs = {}
+
+    if job is None:
+        # use default gejsv values
+        sva, u, v, work, iwork, info = gejsv(A)
+    elif type(job) == list:
+        # execute with modified job argument
+        out = eval("gejsv(A, " + ",".join(list(map("=".join, job))) + ")")
+        sva, u, v, work, iwork, info = out
+        # add each job to dictionary
+        for job_ in job:
+            jobs[job_[0]] = job_[1]
+    else:
+        sva, u, v, work, iwork, info = eval("gejsv(A, " + "=".join(job) + ")")
+
+        jobs[job[0]] = job[1]
 
     assert_equal(info, 0)
 
     SIGMA = np.diag(work[1] / work[0] * sva[:n])
-    sigma = work[1] / work[0] * sva[:n] # not a matrix
+    sigma = work[1] / work[0] * sva[:n]
 
+    assert_allclose(np.sort(sigma), np.sort(svd(A)[0]), atol=atol)
 
-    assert_allclose(np.sort(sigma), np.sort(eig(A)[0]), rtol=rtol, atol=atol)
-    # ^ this doesn't seem like it would work as sva is real and eig(A) is complex
-    assert_allclose(A, u @ SIGMA @ v.T, rtol=rtol, atol=atol)
-    assert_allclose(u.T @ u, np.identity(u.shape[1]), rtol=rtol, atol=atol)
-    assert_allclose(v @ v.T, np.identity(n), rtol=rtol, atol=atol)
+    # run this test if jobu is computed
+    if not jobs.get('jobu') == "'N'":
```

Comment:
@mdhaber It says in the documentation

From the documentation it looks to me that `u` is not usable for this part of the test when `jobu=N` or `W`, but I'm not as sure about F: 

> If JOBU = 'F', then U contains on exit the M-by-M matrix of  the left singular vectors, including an ONB   of the orthogonal complement of the Range(A)

It looks to me that something is going to be returned, but not what will work with this option. 


**Long story short, should I only use `u`  when` jobu = 'U'`?**

Reply:
I think all the singular vectors will still be orthogonal, so they should produce an identity matrix, and the number of rows/columns will depend on `u.shape[1]`, so the answer to your question is _no_: I think this should also work with `jobu == "F"`. I think you're right that this should be skipped if `jobu =="W"`, though. 

Looks like you have both single and double quotes there, BTW.

Reply:
The single quotes inside are intended (clunky, i know)